### PR TITLE
feat(auth): set up basic auth with devise and jwt

### PR DESCRIPTION
**Pull request by mapra99**

### CODE REVIEW

File: api/config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  devise_for :users, defaults: { format: :json }, path: 'api/v1/auth'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  namespace :api do
+    namespace :v1 do
+      resources :test, only: [:index]
```
Comment (ImMPrada) [inline api/config/routes.rb:6]: so this is like `/health` ?

File: api/spec/controllers/api/v1/test_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      before do
+        get :index
+      end
+
+      it 'returns 401' do
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    describe 'when user is authenticated' do
```
Comment (ImMPrada) [inline api/spec/controllers/api/v1/test_controller_spec.rb:15]: Don´t you need to test that the Authorization be returned in the header?

File: api/app/controllers/api/v1/base_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module Api
+  module V1
+    class BaseController < Api::BaseController
+      before_action :authenticate_user!
+    end
+  end
+end
```
Comment (ImMPrada) [inline api/app/controllers/api/v1/base_controller.rb:7]: so, any endpoint's controller son of BaseController will require an authenticated user, right?

**Review by ImMPrada — APPROVED** [review]
nice! left some comments

File: api/config/routes.rb (hunk 1 above)
Comment (mapra99) [reply api/config/routes.rb:6]: Not really. This is just a dummy endpoint I created to test if a protected route worked as expected

File: api/app/controllers/api/v1/base_controller.rb (hunk 3 above)
Comment (mapra99) [reply api/app/controllers/api/v1/base_controller.rb:7]: That's correct. So I'm basically making all my endpoints protected by default. If there is any specific controller or action that needs to be public, I can use `skip_before_action :authenticate_user!` on that controller

File: api/spec/controllers/api/v1/test_controller_spec.rb (hunk 2 above)
Comment (mapra99) [reply api/spec/controllers/api/v1/test_controller_spec.rb:15]: All the authorization header dynamic is something I would test in an acceptance test, maybe in something like postman or cypress

**Pull request merged**